matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/DefaultCryptoService.kt

```diff
@@ -1320,16 +1320,15 @@ internal class DefaultCryptoService @Inject constructor(
         cryptoStore.logDbUsageInfo()
     }
 
-    override fun prepareToEncrypt(roomId: String, callback: MatrixCallback<Unit>) {
+    override suspend fun prepareToEncrypt(roomId: String) {
         cryptoCoroutineScope.launch(coroutineDispatchers.crypto) {
             Timber.tag(loggerTag.value).d("prepareToEncrypt() roomId:$roomId Check room members up to date")
             // Ensure to load all room members
             try {
                 loadRoomMembersTask.execute(LoadRoomMembersTask.Params(roomId))
             } catch (failure: Throwable) {
                 Timber.tag(loggerTag.value).e("prepareToEncrypt() : Failed to load room members")
-                callback.onFailure(failure)
-                return@launch
+                throw failure
```

Comment:
If I've understood correctly, throwing here shouldn't have any behaviour changes as it's the same as `awaitCallback`?

Reply:
I am not sure :/
I will do more test, or I should say I will do test 🙈

Reply:
The app crashes :/

Reply:
@ouchadam I have reverted this change as we (V and G) are not sure about the impact on the crypto code here.

Reply:
thanks for checking! :+1: